sdk/go/kubernetes/helm/v3/chart.go

```diff
@@ -0,0 +1,496 @@
+//                             },
+//                         })
+//                         opts = append(opts, aliases)
+//                     }
+//                 },
+//
+//                 // Omit a resource from the Chart by transforming the specified resource definition
+//                 // to an empty List.
+//                 func(state map[string]interface{}, opts ...pulumi.ResourceOption) {
+//                     name := state["metadata"].(map[string]interface{})["name"]
+//                     if state["kind"] == "Pod" && name == "test" {
+//                         state["apiVersion"] = "core/v1"
+//                         state["kind"] = "List"
+//                     }
+//                 },
+//		       },
+//	       })
+//	       if err != nil {
+//	           return err
+//	       }
+//
+//	       return nil
+//     })
+// }
+// ```
+type Chart struct {
+	pulumi.ResourceState
+
+	Resources pulumi.Output
+}
+
+// NewChart registers a new resource with the given unique name, arguments, and options.
+func NewChart(ctx *pulumi.Context,
+	name string, args ChartArgs, opts ...pulumi.ResourceOption) (*Chart, error) {
+
+	// Register the resulting resource state.
+	chart := &Chart{}
+	err := ctx.RegisterComponentResource("kubernetes:helm.sh/v2:Chart", name, chart, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	// Honor the resource name prefix if specified.
+	if args.ResourcePrefix != "" {
+		name = args.ResourcePrefix + "-" + name
+	}
+
+	resources := args.ToChartArgsOutput().ApplyT(func(args chartArgs) (map[string]pulumi.Resource, error) {
+		return parseChart(ctx, name, args, pulumi.Parent(chart))
+	})
+	chart.Resources = resources
+
+	// Finally, register all of the resources found.
+	// Note: Go requires that we "pull" on our futures in order to get them scheduled for execution. Here, we use
+	// the engine's RegisterResourceOutputs to wait for the resolution of all resources that this Helm chart created.
+	err = ctx.RegisterResourceOutputs(chart, pulumi.Map{"resources": resources})
+	if err != nil {
+		return nil, errors.Wrap(err, "registering child resources")
+	}
+
+	return chart, nil
+}
+
+func parseChart(ctx *pulumi.Context, name string, args chartArgs, opts ...pulumi.ResourceOption,
+) (map[string]pulumi.Resource, error) {
+
+	// Create temporary directory and file to hold chart data and override values.
+	chartDir, err := ioutil.TempDir("", "")
+	if err != nil {
+		return nil, errors.Wrap(err, "creating temp directory for chart")
+	}
+	defer os.RemoveAll(chartDir)
+	overrides, err := ioutil.TempFile("", "values.*.yaml")
+	if err != nil {
+		return nil, errors.Wrap(err, "creating temp file for chart values")
+	}
+	defer os.Remove(overrides.Name())
+
+	var chart string
+	if args.Path != "" { // Local Chart
+		chart = args.Path
+	} else { // Remote Chart
+		if strings.HasPrefix(args.Repo, "http") {
+			return nil, fmt.Errorf("`repo` specifies the name of the Helm chart repo. Use FetchOpts.Repo" +
```

Comment:
Should this be `FetchArgs.Repo`?

Reply:
Fixed in https://github.com/pulumi/pulumi-kubernetes/pull/1211/commits/f889d0b8a419fb1ce852e0584ff6e97a4d3dee38